### ISSUE TITLE
The go-vet checker is incomplete/wrong due to checking only one file

### ISSUE THREAD

**Issue by siebenmann** [flycheck/flycheck#2365]
### Bug description

The go-vet checker is currently incomplete and produces incorrect results because it only checks a single file at a time, which causes `go vet` to complain about cross-file references. Take the following three files, a.go, b.go, and go.mod:

a.go:
```
package testmod

type aStruct struct {
        a int
}
```
b.go:
```
package testmod

func AFunc() *aStruct {
        a := aStruct{}
        return &a
}
```
go.mod:
```
module testmod

go 1.25.0
```
If you run `go vet b.go` it reports `vet: ./b.go:3:15: undefined: aStruct`, although `go vet .` correctly reports no errors. Unfortunately, flycheck runs `go vet` in exactly this way, resulting in spurious error reports. I believe modern `go vet` must be run on the entire project or at least the entire current directory in order to work properly, and intrinsically no longer can run on a single file.

### Steps to reproduce

1. create a directory with the three example files in it.
2. start Emacs to edit b.go and enable flycheck-mode
3. observe the go-vet error reported

### Expected behavior

No error from go-vet.

### Output of M-x flycheck-verify-setup

```text
Syntax checkers for buffer b.go in go-mode:

First checker to run:

  go-gofmt
    - may enable:    yes
    - executable:    Found at /u/cks/bin/bin.i386-linux/gofmt
    - next checkers: go-vet, go-build, go-test, go-errcheck, go-unconvert, go-staticcheck

Checkers that may run as part of the first checker's chain:

  go-vet
    - may enable:    yes
    - executable:    Found at /u/cks/bin/bin.i386-linux/go
    - go tool vet:   present
    - next checkers: go-build, go-test, go-errcheck, go-unconvert, go-staticcheck

  go-build
    - may enable:    yes
    - may run:       t
    - executable:    Found at /u/cks/bin/bin.i386-linux/go
    - next checkers: go-errcheck, go-unconvert, go-staticcheck

  go-errcheck
    - may enable:    yes
    - may run:       t
    - executable:    Found at /homes/hawkwind/u0/cks/go/bin/errcheck
    - next checkers: go-unconvert, go-staticcheck

  go-unconvert
    - may enable: yes
    - may run:    t
    - executable: Found at /homes/hawkwind/u0/cks/go/bin/unconvert

  go-staticcheck
    - may enable: yes
    - executable: Found at /homes/hawkwind/u0/cks/go/bin/staticcheck

Checkers that are compatible with this mode, but will not run until properly configured:

  go-test
    - may enable:    yes
    - may run:       nil
    - executable:    Found at /u/cks/bin/bin.i386-linux/go
    - next checkers: go-errcheck, go-unconvert, go-staticcheck

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: 20260824.908
Emacs version:    31.1.50
System:           x86_64-pc-linux-gnu
Window system:    nil
```

### Emacs version

GNU Emacs 31.1.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.52, cairo version 1.18.4) of 2026-08-24

### Emacs configuration

Plain Emacs / Custom configuration

### Additional notes

This reproduces in `emacs -Q`, and I've verified through external program execution tracing that `go vet` is run on only a single file, `b.go`.
